eos/constraint.cc

```diff
@@ -1652,176 +1654,176 @@ namespace eos
                     }
                 }
 
+                std::vector<QualifiedName> observables;
                 try
                 {
-                    std::vector<QualifiedName> observables;
                     for (auto && o : n["observables"])
                     {
                         observables.push_back(QualifiedName(o.as<std::string>()));
                     }
+                }
+                catch (QualifiedNameSyntaxError & e)
+                {
+                    throw ConstraintDeserializationError(name, "'" + observables.back().full() + "' is not a valid observable name (" + e.what() + ")");
```

Comment:
`back` is incorrect since the observable has not been pushed to the list yet.

Reply:
Indeed! Fixed.